atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/TransactionKeyValueService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api;
+
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+import java.util.Map;
+
+public interface TransactionKeyValueService extends AsyncTransactionKeyValueService {
+    Map<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>> getFirstBatchForRanges(
```

Comment:
@jeremyk-91 it's very interesting that this is all of the API that transactions use. I might've missed something, currently struggling to run compilation locally in this repo...